message_adapter/message_adapter.py

```diff
@@ -109,10 +109,15 @@ def loadAndUpdateRemoteEvent(self, event, context):
         * @returns {*} the full event data
         """
         if ('replace' in event):
+            exception = None
+            if ('exception' in event):
+                exception = event['exception']
             _s3 = s3()
             data = _s3.Object(event['replace']['Bucket'], event['replace']['Key']).get()
             if (data is not None):
                 event = json.loads(data['Body'].read().decode('utf-8'))
+                if (exception is not None):
```

Comment:
If the remote message already has exception, why do we need to assign the exception to the event retrieved again?

Reply:
Otherwise it is lost when event is overwritten by data.

Reply:
Is it true that the short message (with or without 'replace') and remote message both have the 'exception' if there is a workflow error?  Then the data from s3 should have the exception already.

Reply:
My experience here (when I was doing some EDPA development using the CMA) is that, in the current implementation, the exception is in the message will exist in the output of a state that errors out, but that exception doesn't exist in the remote (S3) file. I'm looking through some of my previous executions and the remote message shows `exception: 'None'`.

Reply:
The sample in the CUMULUS-860 ticket is actual output I received from MoveGranules. I'm not really sure how or when the exception is caught and put into the message, but it does not show up in the remote message, and does show up in the short message.

Reply:
Handling the exception that occurs otherwise will depend on the client libraries.

Reply:
I understand the remote message doesn't have error exception currently.  But this PR has put the error exception to s3 message, so the code doesn't need to worry about any more.